src/runtime/thread_storage_scope.h

```diff
@@ -57,6 +57,8 @@ enum class StorageRank {
   kWMMAMatrixB = 5,
   /*! \brief wmma scope memory of accumulator */
   kWMMAAccumulator = 6,
+  /*! \brief global scope texture memory */
+  kTexture = 7,
```

Comment:
I believe we do not need to introduce a special scope. Instead we can directly make texture a tag, for example: ```global.texture-weight```

This would reduce special casing in thread hierachy calculations. We should actually also remove WMMA later